### ISSUE TITLE
Charge timer and charge limit no longer working

### ISSUE THREAD

**Issue by seb43654** [dalathegreat/Nissan-env200-Battery-Upgrade#4]
After upgrading from 24kwh to 40kwh, the 80% charge limit and the charge timer no longer seem to function. Have these been successfully tested by someone else with this code running?

**Comment by glynhudson** [comment]
Yes, I can confirm the dashboard 80% charge limit doesn't work on my upgraded 40kWh e-NV200 either. I've not tested the timer I've never used it. I presume the 80% charge limit was a feature of the 24kWh battery, since stock 40kWh e-NV200 do not have an 80% charge limit either. 

However, not having a charge limit is not a big issue since OVMS now supports remotely stopping a charge and setting a charge limit.

**Comment by dalathegreat** [comment]
I can confirm, that the newer battery packs don't understand the 80% longlife option at all. However, it would be possible to code this into the CAN-bridge for users that don't have OVMS.

The 0x1F2 message sent by the VCM contains a parameter called "TargetChargeSOC". This bit has two values, 0=100% and 1=80%. In theory, we could quite simply intercept this, and force a charge stop via the same mechanic as OVMS does.

Seb / Glyn, incase anyone of you wants to test this, let me know and I can code up a test.

**Comment by seb43654** [comment]
I would be happy to test this since my Mums van doesnt have OVMS and she wouldnt want to add it. Only catch is the van is up in Scotland, so I wouldnt be able to apply the SW update until I go and see her next (likely around 1 months time) However if your able to add the software feature in I can flash it as soon as i am up with the van. Also any idea why the charge timer doesnt work?

**Comment by glynhudson** [comment]
> Seb / Glyn, incase anyone of you wants to test this, let me know and I can code up a test.

Sure, I'm happy to help test. Getting rid of the 80% limit was a silly move by Nissan IMO

**Comment by dalathegreat** [comment]
I created a branch for this feature (https://github.com/dalathegreat/Nissan-env200-Battery-Upgrade/commit/3417a4dbfa320f73d8f1021423add45b687f41c6)

**Comment by seb43654** [comment]
Thanks I will provide feedback when I can test it!

**Comment by dalathegreat** [comment]
We 100% fixed this on the LEAF upgrades, here is the same fix applied to env200! https://github.com/dalathegreat/Nissan-env200-Battery-Upgrade/pull/7

**Comment by andyorkney** [comment]
> Yes, I can confirm the dashboard 80% charge limit doesn't work on my upgraded 40kWh e-NV200 either. I've not tested the timer I've never used it. I presume the 80% charge limit was a feature of the 24kWh battery, since stock 40kWh e-NV200 do not have an 80% charge limit either. 
> 
> 
> 
> However, not having a charge limit is not a big issue since OVMS now supports remotely stopping a charge and setting a charge limit. 

So how do we set charge limit in OVMS when I cannot get OVMS app to work, there must be a script action/command but I can't find documentation for it

**Comment by glynhudson** [comment]
The OVMS command is `config set xnl suffsoc 80`

This will set the SoC limit to 80%. Chance this to whatever limit you want.

**Comment by andyorkney** [comment]
> The OVMS command is `config set xnl suffsoc 80`
> 
> This will set the SoC limit to 80%. Chance this to whatever limit you want. 

Ok that's really helpful Glyn, I've been trying to implement a script to charge in cheap window (we're using granny charger) and stop at user set SOC and I have had that working using ticker-based monitoring.  So now I'm looking at suffsoc as a way forward but it did not stop the charge just developed an alert of suffsoc reached. So now not sure how it will actually help in charge control? What am I missing? Thanks

**Comment by andyorkney** [comment]
> > The OVMS command is `config set xnl suffsoc 80`
> > 
> > This will set the SoC limit to 80%. Chance this to whatever limit you want. 
> 
> Ok that's really helpful Glyn, I've been trying to implement a script to charge in cheap window (we're using granny charger) and stop at user set SOC and I have had that working using ticker-based monitoring.  So now I'm looking at suffsoc as a way forward but it did not stop the charge just developed an alert of suffsoc reached. So now not sure how it will actually help in charge control? What am I missing? Thanks

Doh! Didn't set Autocharge. Now have success with suffsoc and autocharge set. Thank you
